Docs: Mention explicit_indirect_conditions & "Menu" -> origin_region_name (#3887)
ArchipelagoMW/Archipelago: docs/world api.md

Regions have a list called `exits`, containing `Entrance` objects representing transitions to other regions.

> There must be one special region (Called "Menu" by default, but configurable using [origin_region_name](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L295-L296)),

Reviewer: I worry about this link since it will have to be updated every single time anything changes in AutoWorld

Author: True, not sure what else to do tho

Reviewer: Fair enough LOL

You can use `multiworld.register_indirect_condition(region, entrance)` to explicitly tell the generator that, when a given region becomes accessible, it is necessary to re-check a specific entrance.
You **must** use `multiworld.register_indirect_condition` if you perform this kind of `can_reach` from an entrance access rule, unless you have a **very** good technical understanding of the relevant code and can reason why it will never lead to problems in your case.

> Alternatively, you can set [world.explicit_indirect_conditions = False](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L298-L301),
> in which case your world takes a flat performance hit to not have to worry about indirect conditions.

Reviewer 2: ```suggestion
Alternatively, you can set [world.explicit_indirect_conditions = False](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L298-L301),
avoiding the need for indirect conditions at the expense of performance.
```
This line didn't make sense until I read the docstring that was linked to. Tried to reword to better convey that message, but I don't have any strong feelings about this change.
